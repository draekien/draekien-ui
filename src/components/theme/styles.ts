export const styles = {
  root: {
    fontFamily: 'Montserrat, sans-serif',
    variant: 'text.medium',
    '*:focus': {
      outline: 'none',
    },
  },
};
