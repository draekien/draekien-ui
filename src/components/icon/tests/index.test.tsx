/** @jsx jsx */
import { render } from '@testing-library/react';
import { jsx } from 'theme-ui';
import Icon from '..';
import { iconCss } from '../index.styles';

describe('Icon component', () => {
  test('default icon matches snapshot', () => {
    const component = render(<Icon name="accessibility" />);

    expect(component.container).toMatchSnapshot();
  });

  test('outlined icon matches snapshot', () => {
    const component = render(<Icon name="accessibility" variant="outlined" />);

    expect(component.container).toMatchSnapshot();
  });
});

describe('iconCss function', () => {
  test('returns expected css when size is medium', () => {
    const expectedCss = {
      verticalAlign: 'middle',
      color: 'inherit',
      fontSize: '1.5rem',
    };

    expect(iconCss({ size: 'medium' })).toEqual(expectedCss);
  });

  test('returns expected css when size is small', () => {
    const expectedCss = {
      verticalAlign: 'middle',
      color: 'inherit',
      fontSize: '1rem',
    };

    expect(iconCss({ size: 'small' })).toEqual(expectedCss);
  });

  test('returns expected css when size is large', () => {
    const expectedCss = {
      verticalAlign: 'middle',
      color: 'inherit',
      fontSize: '2rem',
    };

    expect(iconCss({ size: 'large' })).toEqual(expectedCss);
  });

  test('returns expected css when color is primary', () => {
    const expectedCss = {
      verticalAlign: 'middle',
      color: 'primary',
      fontSize: '1.5rem',
    };

    expect(iconCss({ size: 'medium', color: 'primary' })).toEqual(expectedCss);
  });
});
