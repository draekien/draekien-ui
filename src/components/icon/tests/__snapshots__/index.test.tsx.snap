// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component default icon matches snapshot 1`] = `
<div>
  <i
    class="material-icons  css-eyhhi6"
  >
    accessibility
  </i>
</div>
`;

exports[`Icon component outlined icon matches snapshot 1`] = `
<div>
  <i
    class="material-icons-outlined  css-eyhhi6"
  >
    accessibility
  </i>
</div>
`;
